.github/dependabot.yml

```diff
@@ -13,18 +13,87 @@ updates:
     # For each dependencies group, dependabot will open a PR containing
     # the new versions available for that group.
     groups:
-      # wgpu dependencies group
-      wgpu-dependencies:
+      # all dependencies group
+      all-dependencies:
         patterns:
+        - "arboard"
+        - "async-trait"
+        - "bincode"
+        - "blas-src"
+        - "bytemuck"
+        - "candle-core"
+        - "clap"
+        - "console_error_panic_hook"
+        - "crossterm"
+        - "csv"
+        - "dashmap"
+        - "derive-new"
+        - "dirs"
+        - "fake"
+        - "flate2"
+        - "float-cmp"
         - "futures-intrusive"
+        - "getrandom"
+        - "github-device-flow"
+        - "gix-tempfile"
```

Comment:
I would very much like to not duplicate out dependencies outside of `Cargo.toml`, otherwise this list isn't going to be comprehensive and will for sure become outdated quite soon.

Reply:
Yep, I was thinking the same, but let's list the possible actions we can have:

- If we remove a dep then it is a nop for `.dependabot` since it cannot find that dep anymore. The dependency name can stay there without affecting anything.
- If we change or add a new dependency to `Cargo.toml`, an independent PR is created for that dependency because it is not part of the group. We can later decide to add that dependency to the `all` group, thus removing the independent PR.

We can also simplify this process adding a `xtask` task which updates `.dependabot` file using `Cargo.toml` dependencies and replace the current "aggregation PR" action with this new action which runs occasionally

Reply:
Can we drop the groups and use the combined PR action ? Seems to be less hassle than maintaining a list of dependencies.

Reply:
Agree, that would be perfect. We can scheduled dependabot just before the combine PR.

Reply:
Done